In [ ]:
from tensorflow.keras.preprocessing.text import one_hot


In [ ]:
### sentences
sent=[  'the glass of milk',
     'the glass of juice',
     'the cup of tea',
    'I am a good boy',
     'I am a good developer',
     'understand the meaning of words',
     'your videos are good',]

### Vocabulary size

In [ ]:

voc_size=10000

####Convert Sentences into one-hot Representation

####In this basically we can see every words in a sentence and give each word to one hot function that give the index of every word in vocab_size

In [ ]:
onehot_repr=[one_hot(words,voc_size)for words in sent] 
print(onehot_repr)


[[6961, 9615, 4954, 1615], [6961, 9615, 4954, 3253], [6961, 5953, 4954, 8738], [6049, 2542, 7807, 9280, 446], [6049, 2542, 7807, 9280, 776], [6204, 6961, 3696, 4954, 7582], [5447, 6659, 5550, 9280]]


###Here we See above one_hot representation the out of 10000 vocab_size the word the index at $6961$

###Next step is to pass the one-hot into our embedding layers present in keras

####To form a embeddding matrix

In [ ]:
from tensorflow.keras.layers import Embedding
from tensorflow.keras.preprocessing.sequence import pad_sequences##it will do basicaly the same size of each and every sentences
from tensorflow.keras.models import Sequential##to create embedding layer model itself

In [ ]:
import numpy as np


###padding basicallly do that it can covert the sent into its default to sent_length
###by giving pre value zero and last remaning value is our sentence

In [ ]:
sent_length=8##user give your own chooice of sentence length
embedded_layer=pad_sequences(onehot_repr,padding='pre',maxlen=sent_length)
print(embedded_layer)

[[   0    0    0    0 6961 9615 4954 1615]
 [   0    0    0    0 6961 9615 4954 3253]
 [   0    0    0    0 6961 5953 4954 8738]
 [   0    0    0 6049 2542 7807 9280  446]
 [   0    0    0 6049 2542 7807 9280  776]
 [   0    0    0 6204 6961 3696 4954 7582]
 [   0    0    0    0 5447 6659 5550 9280]]


In [ ]:
dim=10

In [ ]:
model=Sequential()
model.add(Embedding(voc_size,10,input_length=sent_length))
model.compile('adam','mse')

In [ ]:
model.summary()

Model: "sequential"
_________________________________________________________________
Layer (type)                 Output Shape              Param #   
embedding (Embedding)        (None, 8, 10)             100000    
Total params: 100,000
Trainable params: 100,000
Non-trainable params: 0
_________________________________________________________________


###Now seee we converted Our Embdeed layers  into featurised vectors Something Look like Below

In [ ]:
print(model.predict(embedded_layer))

[[[ 2.03479081e-04 -1.80694573e-02 -3.74273770e-02  1.66234709e-02
    1.12777576e-02  2.36534514e-02  3.29170339e-02 -1.64546072e-04
    3.76234390e-02  4.98418547e-02]
  [ 2.03479081e-04 -1.80694573e-02 -3.74273770e-02  1.66234709e-02
    1.12777576e-02  2.36534514e-02  3.29170339e-02 -1.64546072e-04
    3.76234390e-02  4.98418547e-02]
  [ 2.03479081e-04 -1.80694573e-02 -3.74273770e-02  1.66234709e-02
    1.12777576e-02  2.36534514e-02  3.29170339e-02 -1.64546072e-04
    3.76234390e-02  4.98418547e-02]
  [ 2.03479081e-04 -1.80694573e-02 -3.74273770e-02  1.66234709e-02
    1.12777576e-02  2.36534514e-02  3.29170339e-02 -1.64546072e-04
    3.76234390e-02  4.98418547e-02]
  [ 1.97069310e-02  9.89841297e-03  3.52151282e-02 -2.91153789e-02
    3.45156305e-02  4.16117422e-02 -2.10133791e-02 -3.24141532e-02
    2.84065045e-02 -1.92613360e-02]
  [-8.92989337e-05 -1.05079263e-03  1.87988207e-03 -3.24426293e-02
   -1.15262382e-02  1.62631534e-02  4.74228524e-02 -1.67931095e-02
    4.89989258e-

##So Basically what we see here is that 
##supoose if one word index is $3000$ <---- so it will convert it into dimension of $10$ vectors
###Same here is happening $0$ is converted into  $---->$

$[ 2.0347908e-04 -1.8069457e-02 -3.7427377e-02  1.6623471e-02
   1.1277758e-02  2.3653451e-02  3.2917034e-02 -1.6454607e-04
   3.7623439e-02  4.9841855e-02]$

###into $10 $ vectors  dimensions

In [ ]:
embedded_layer[0]


array([   0,    0,    0,    0, 6961, 9615, 4954, 1615], dtype=int32)

###Embedding matrix

In [ ]:
print(model.predict(embedded_layer)[0])


[[ 2.0347908e-04 -1.8069457e-02 -3.7427377e-02  1.6623471e-02
   1.1277758e-02  2.3653451e-02  3.2917034e-02 -1.6454607e-04
   3.7623439e-02  4.9841855e-02]
 [ 2.0347908e-04 -1.8069457e-02 -3.7427377e-02  1.6623471e-02
   1.1277758e-02  2.3653451e-02  3.2917034e-02 -1.6454607e-04
   3.7623439e-02  4.9841855e-02]
 [ 2.0347908e-04 -1.8069457e-02 -3.7427377e-02  1.6623471e-02
   1.1277758e-02  2.3653451e-02  3.2917034e-02 -1.6454607e-04
   3.7623439e-02  4.9841855e-02]
 [ 2.0347908e-04 -1.8069457e-02 -3.7427377e-02  1.6623471e-02
   1.1277758e-02  2.3653451e-02  3.2917034e-02 -1.6454607e-04
   3.7623439e-02  4.9841855e-02]
 [ 1.9706931e-02  9.8984130e-03  3.5215128e-02 -2.9115379e-02
   3.4515630e-02  4.1611742e-02 -2.1013379e-02 -3.2414153e-02
   2.8406505e-02 -1.9261336e-02]
 [-8.9298934e-05 -1.0507926e-03  1.8798821e-03 -3.2442629e-02
  -1.1526238e-02  1.6263153e-02  4.7422852e-02 -1.6793109e-02
   4.8998926e-02  4.0595893e-02]
 [-1.5962504e-02  7.0470199e-03 -2.4171924e-02 -1.8063914e